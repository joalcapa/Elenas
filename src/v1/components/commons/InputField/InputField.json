{
  "description": "Este componente renderiza un input basico.",
  "displayName": "InputField",
  "methods": [],
  "props": {
    "value": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "placeholder": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "onChangeText": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "isPassword": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}