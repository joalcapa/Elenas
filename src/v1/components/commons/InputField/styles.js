import {StyleSheet} from 'react-native';

import {Colors} from '../../../resources';

const styles = StyleSheet.create({
    input: {
        width: '100%',
        maxWidth: 150,
        paddingTop: 5,
        paddingBottom: 5,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5,
        borderWidth: 1,
        borderColor: Colors.C_7e807f,
    },
});

export default styles;