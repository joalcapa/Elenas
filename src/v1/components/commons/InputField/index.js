import React from 'react';
import PropTypes from 'prop-types';
import {TextInput} from 'react-native';

import styles from './styles';

/**
 * Este componente renderiza un input basico.
 */
const InputField = (props) => 
    <TextInput 
        {...props} 
        style={styles.input}
        secureTextEntry={props.isPassword}
    />;

InputField.propTypes = {
    value: PropTypes.string,
    placeholder: PropTypes.string,
    onChangeText: PropTypes.func,
    isPassword: PropTypes.bool,
};

InputField.defaultProps = {
    /**
     * Valor del input.
     */
    value: '',
    /**
     * Informacion que identifica al input.
     */
    placeholder: '',
    /**
     * Funcion que modifica el valor del input.
     */
    onChangeText: () => {},
    /**
     * Valor que indica si es input de contrase√±a.
     */
    isPassword: false,
};

export default InputField;