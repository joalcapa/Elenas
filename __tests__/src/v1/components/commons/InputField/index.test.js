import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {create} from 'react-test-renderer';
import {TextInput} from 'react-native';
import {shallow, configure} from 'enzyme';

import InputField from '../../../../../../src/v1/components/commons/InputField';

configure({adapter: new Adapter()});
describe('InputField component', () => {
    test('render InputField component', () => {
        const tree = create(<InputField />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render InputField component with value', () => {
        const tree = create(<InputField value="input"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render InputField component with placeholder', () => {
        const tree = create(<InputField placeholder={'placeholder text'}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onChangeText functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(<InputField onChangeText={onPressEvent} />);

        wrapper.find(TextInput).first().props().onChangeText('value');
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });
});