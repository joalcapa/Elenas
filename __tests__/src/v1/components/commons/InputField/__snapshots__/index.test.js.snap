// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component render InputField component 1`] = `
<input
  autoCapitalize="sentences"
  autoComplete="on"
  autoCorrect="on"
  className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
  dir="auto"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onSelect={[Function]}
  placeholder=""
  readOnly={false}
  spellCheck={true}
  type="text"
  value=""
/>
`;

exports[`InputField component render InputField component with placeholder 1`] = `
<input
  autoCapitalize="sentences"
  autoComplete="on"
  autoCorrect="on"
  className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
  dir="auto"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onSelect={[Function]}
  placeholder="placeholder text"
  readOnly={false}
  spellCheck={true}
  type="text"
  value=""
/>
`;

exports[`InputField component render InputField component with value 1`] = `
<input
  autoCapitalize="sentences"
  autoComplete="on"
  autoCorrect="on"
  className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
  dir="auto"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onSelect={[Function]}
  placeholder=""
  readOnly={false}
  spellCheck={true}
  type="text"
  value="input"
/>
`;
